cross_validation_mpi.py

```diff
@@ -0,0 +1,189 @@
+#from sklearn.ensemble import GradientBoostingClassifier
+#from sklearn.discriminant_analysis import QuadraticDiscriminantAnalysis
+#from sklearn.naive_bayes import GaussianNB as GaussianNB
+#from sklearn.discriminant_analysis import QuadraticDiscriminantAnalysis as QuadraticDiscriminantAnalysis
+#from sklearn.neighbors import KNeighborsClassifier as KNeighborsClassifier
+#from sklearn.pipeline import Pipeline
+#from sklearn.decomposition import PCA
+#from sklearn.tree import DecisionTreeClassifier as DecisionTreeClassifier
+
+DATASET_DIR = '/users/data/d3m/datasets/training_datasets'
+METADATA_PATH = '~/data/closed_d3m_unembed_data.csv'
+  
+def save_results(results,conf):
+    #save the results to a csv file
+    results.to_csv(model_name+'_final_cross_val.csv',index=False)
+    filename = model_name+'_matrix_mean.pkl'
+    fileObject = open(filename, 'wb')
+    pickle.dump(conf, fileObject)
+    fileObject.close()
+    
+def naive_gen():
+    model_name = 'Naive'
+    class NaiveModel:
+        def fit(self,X_train,y_train):
+            self.majority = y_train.value_counts().idxmax()
+        def predict(self,X_test):
+            y_hat = [self.majority for i in range(len(X_test))]
+            return y_hat
+    model = NaiveModel()
+    return model, model_name
+
+def evaluate_model(balance: bool, col_name: bool, use_metadata: bool, rank=None):
+
+    def run_fold(train_ind, test_ind, balance=True):
+        #now fit using the indeces given by the kfold splitter
+        X_train = X_data[train_ind]
+        y_train = y.iloc[train_ind]
+        #get the labels for the confusion matrix
+        if (balance == True):
+            #get the k_neighbors balance number
+            k_neighbors = y_train.value_counts().min()-1
+            assert k_neighbors > 0, 'Not enough data to rebalance. Must be more than 1:.'
+            #rebalance
+            #print("balancing")
+            begin = time.time()
+            #smote = ADASYN(sampling_strategy='not majority',n_neighbors=k_neighbors,random_state=32)
+            #smote = SMOTE(sampling_strategy='not majority',k_neighbors=k_neighbors,random_state=32)
+            smote = BorderlineSMOTE(sampling_strategy='not majority',k_neighbors=k_neighbors,random_state=32)
+
+            #smote = SVMSMOTE(sampling_strategy='not majority',k_neighbors=k_neighbors,random_state=32)
+            X_train, y_train = smote.fit_resample(X_train,y_train)
+            end = time.time()
+            print("Time to rebalance: "+str(np.round(end-begin,3)))       
+        #fit on  data
+        model.fit(X_train,y_train)
+        #predict on the model
+        del X_train
+        del y_train
+        y_hat = list(model.predict(X_data[test_ind]))
+        y_test = list(y.iloc[test_ind])
+        print("Finished Fold "+str(rank))
+        return y_hat, y_test
+  
+    if (rank == 0):
+        print("Beginning cross validation")
+        if (use_metadata):
+            if (col_name is True):
```

Comment:
I don't see a difference between this `if` block and the `else` block.

Reply:
The difference is in the filename, the one is a file with only embedded column names and the other contains all 3 metadata.